main.py

```diff
@@ -53,7 +53,131 @@ def handle_arguments(cl_arguments):
+                             strict=strict)
+        else:
+            assert_for_log(args.allow_untrained_encoder_parameters,
+                           "No best checkpoint found to evaluate.")
+            log.warning("Evaluating untrained encoder parameters!")
+    return task_names_to_avoid_loading, strict
+
+def check_configeration(args, pretrain_tasks, target_tasks):
+    """
+    Checks configerations for any obvious logical flows
+    and that necessary parameters are set for each step - 
+    throws asserts and exits if found.
+
+    Parameters
+    ----------------
+    args: Params object
+    pretrain_tasks: list of pretraining Task objects
+    target_tasks: list of target task training Task objects
+
+    Returns
+    ----------------
+    None
+    """
+    steps_log = []
+
+    if not args.load_eval_checkpoint == 'none':
+        assert_for_log(os.path.exists(args.load_eval_checkpoint),
+                       "Error: Attempting to load model from non-existent path: [%s]" %
+                       args.load_eval_checkpoint)
+        assert_for_log(
+            not args.do_pretrain,
+            "Error: Attempting to train a model and then replace that model with one from a checkpoint.")
+        steps_log.append("Loading model from path: %s" % args.load_eval_checkpoint)
+
+    assert_for_log(args.transfer_paradigm in ["finetune", "frozen"],
+                   "Transfer paradigm %s not supported!" % args.transfer_paradigm)
+
+    if args.do_pretrain:
+        assert_for_log(args.pretrain_tasks != "none",
+                       "Error: Must specify at least on training task: [%s]" % args.pretrain_tasks)
+        assert_for_log(
+            args.val_interval %
+            args.bpp_base == 0, "Error: val_interval [%d] must be divisible by bpp_base [%d]" %
+            (args.val_interval, args.bpp_base))
+        steps_log.append("Training model on tasks: %s" % args.pretrain_tasks)
+
+    if args.do_target_task_training:
+        steps_log.append("Re-training model for individual eval tasks")
+        assert_for_log(
+            args.eval_val_interval %
+            args.bpp_base == 0, "Error: eval_val_interval [%d] must be divisible by bpp_base [%d]" %
+            (args.eval_val_interval, args.bpp_base))
+        assert_for_log(len(set(pretrain_tasks).intersection(target_tasks)) == 0
+                       or args.allow_reuse_of_pretraining_parameters
+                       or args.do_pretrain == 0,
+                       "If you're pretraining on a task you plan to reuse as a target task, set\n"
+                       "allow_reuse_of_pretraining_parameters = 1(risky), or train in two steps:\n"
+                       "  train with do_pretrain = 1, do_target_task_training = 0, stop, and restart with\n"
+                       "  do_pretrain = 0 and do_target_task_training = 1.")
+
+    if args.do_full_eval:
+        assert_for_log(args.target_tasks != "none",
```

Comment:
I actually don't understand this. What if I want to do a full evaluation on the test sets of my pretraining tasks?

Reply:
We should actually have a eval_tasks list too, since right now, if you want to do what you said, we have to do another run where you create load_eval_checkpoint to a pretraining task, make the target_tasks the pretrain task you want to do an evaluation on. (I actually haven't tried this myself, but this is what I would think you'd have to do).

Reply:
I think that's right but I just find it annoying and not clear why it's the case (@sleepinyourhat do you remember why we require this?).

Reply:
I'll make an issue for this